revolutionpi/analog_pin.go

```diff
@@ -82,6 +82,42 @@ func (b *revolutionPiBoard) WriteAnalog(ctx context.Context, pin string, value i
 	return analogPin.ControlChip.writeValue(int64(analogPin.Address), buf.Bytes())
 }
 
+func (pin *analogPin) Write(ctx context.Context, value int, extra map[string]interface{}) error {
+	pin.ControlChip.logger.Debugf("Analog: %#v", pin)
+	if !pin.isAnalogOutput() {
+		return fmt.Errorf("cannot Write to Analog, pin %s is not an analog output pin", pin.Name)
+	}
+
+	// check to see if analog output is enabled
+	bufOutputRange := make([]byte, 1)
+	outputRangeAddress := pin.inputOffset + 69
+	// use the corresponding analog OutputRange pin to check if the analog output is enabled
+	if pin.Address == pin.outputOffset+2 {
+		outputRangeAddress = pin.inputOffset + 79
+	}
+	n, err := pin.ControlChip.fileHandle.ReadAt(bufOutputRange, int64(outputRangeAddress))
+	if err != nil {
+		return err
+	}
+	if n != 1 {
+		return fmt.Errorf("unable to determine if pin %s is configured for analog write", pin.Name)
+	}
+	pin.ControlChip.logger.Debugf("outputRange Value: %d", bufOutputRange)
+	// at a later date we can use this address to help validate the requested value is within the range of the pin.
+	// for now all we need to check is if the voltage range is not configured.
+	// this results in analog output not being enabled.
+	if bufOutputRange[0] == 0 {
+		return fmt.Errorf("pin %s is not configured for analog write", pin.Name)
+	}
+
+	buf := new(bytes.Buffer)
+	err = binary.Write(buf, binary.LittleEndian, int32(value))
+	if err != nil {
+		return err
+	}
+	return pin.ControlChip.writeValue(int64(pin.Address), buf.Bytes())
```

Comment:
so these are stored as non int64s initially because the [initial structures](https://github.com/viam-soleng/viam-revolution-pi/blob/5850898405466c359613021be8d67c1220abfb40/revolutionpi/gpio_chip.go#L32) the fields come from were non int64 fields. But I also kinda agree that since I have to convert most of these fields to int64 anyway, that I might as well store them in my own structs as int64.

Reply:
I am going to save this change for its own cleanup related PR, that covers all of the files

Reply:
>  save this change for its own cleanup related PR

Excellent choice!